pallets/msa/src/benchmarking.rs

```diff
@@ -101,30 +101,34 @@ benchmarks! {
 	}: _ (RawOrigin::Signed(caller.clone()))
 	verify {
 		assert!(Msa::<T>::get_msa_by_public_key(caller).is_some());
+		assert_eq!(frame_system::Pallet::<T>::events().len(), 1);
 	}
 
 	create_sponsored_account_with_delegation {
+		let s in 0 .. T::MaxSchemaGrantsPerDelegation::get();
 
 		let caller: T::AccountId = whitelisted_caller();
 		assert_ok!(Msa::<T>::create(RawOrigin::Signed(caller.clone()).into()));
 		assert_ok!(Msa::<T>::create_provider(RawOrigin::Signed(caller.clone()).into(),Vec::from("Foo")));
 
-		let (payload, signature, key) = create_payload_and_signature::<T>();
+		let schemas: Vec<SchemaId> = (0 .. s as u16).collect();
+		T::SchemaValidator::set_schema_count(schemas.len().try_into().unwrap());
```

Comment:
It is weird that there is no relationship between schema count in weights. I would probably replace this with the following and rerun to ensure.
https://github.com/LibertyDSNP/frequency/blob/main/pallets/messages/src/benchmarking.rs#L67-L69

Reply:
ahh yeah. I am waiting on the weights to be generated for this on Jenkins.

Reply:
I will wait for the Jenkins build to complete, so it does not restart. However, you should expect to see the following update:

```
#[pallet::weight(T::WeightInfo::create_sponsored_account_with_delegation(
			T::MaxSchemaGrantsPerDelegation::get()
		))]
pub fn create_sponsored_account_with_delegation(...) {}
```